core/debug/src/debug_box.c

```diff
@@ -92,9 +92,7 @@ void debug_halt_error(THaltError reason)
     if (!g_debug_box.initialized || debugged_once_before) {
         while (1);
     } else {
-        /* Remember that debug_deprivilege_and_return() has been called once.
-         * We'll reboot after the debug handler is run, so this will go back to
-         * zero after the reboot. */
+        /* Remember that debug_deprivilege_and_return() has been called once. */
```

Comment:
Perhaps note that the debug box can only be entered once, to avoid the debug box calling itself recursively. This is probably important enough to go into the documentation (not just comments) as well, as this limits what you can do after the debug box runs before you have to reboot. For instance, you can't catch a fault and then restart that box and catch a subsequent fault; you get one debug box entry per boot. This means you pretty much have to reboot.